modules/openapi-generator/src/main/resources/typescript-fetch/runtime.mustache

```diff
@@ -121,7 +121,7 @@ export class BaseAPI {
             // do not handle correctly sometimes.
             url += '?' + this.configuration.queryParamsStringify(context.query);
         }
-        const body = ((typeof FormData !== "undefined" && context.body instanceof FormData) || context.body instanceof URLSearchParams || isBlob(context.body))
+        const body = ((typeof FormData !== "undefined" && context.body instanceof FormData) || context.body instanceof URLSearchParams || isBlob(context.body)) || context.headers['Content-Type'] !== 'application/json'
```

Comment:
please move this check to a separate function (see e.g. https://github.com/OpenAPITools/openapi-generator/blob/2df68d9359a3d8f7d2b2a98ccf4c18546ce0fc9b/modules/openapi-generator/src/main/resources/typescript-angular/configuration.mustache#L139-L152)

one question: is the content-type always set here? maybe we should check for undefined as well

Reply:
I have added the extra function analogous to the angular generator. I also copied the regex magic; I hope that is OK ^^